### PR TITLE
Ignore error message and description when comparing errors lists in model migration tests

### PR DESCRIPTION
**Pull request by piotrp**
Sometimes error messages and descriptions may differ, and then they are reported as new ones in migration tests, even though they are not "new" (caused by migration):
* when description or message in Nu changes
* when locale between local and remote environment differs
* when error message or description depends on runtime environment's settings

### CODE REVIEW

File: components/openapi/src/test/scala/pl/touk/nussknacker/openapi/BaseOpenAPITest.scala [hunk 1]
```diff
@@ -35,7 +36,7 @@ trait BaseOpenAPITest {
   }
 
   protected def parseResource(name: String): String =
-    IOUtils.toString(getClass.getResourceAsStream(s"/swagger/$name"), "UTF-8")
+    IOUtils.toString(getClass.getResourceAsStream(s"/swagger/$name"), StandardCharsets.UTF_8)
```
Comment (lciolecki) [inline components/openapi/src/test/scala/pl/touk/nussknacker/openapi/BaseOpenAPITest.scala:39]: Please use ResourceLoader instead of this.

**Review by lciolecki — APPROVED** [review]
LGTM

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/TestModelMigrations.scala [hunk 2]
```diff
@@ -50,30 +50,36 @@ class TestModelMigrations(migrations: ProcessingTypeDataProvider[ProcessMigratio
   }
 
   private def extractNewErrors(before: ValidationResult, after: ValidationResult) : ValidationResult = {
+    // simplified comparison key: we ignore error message and description
+    def errorToKey(error: NodeValidationError) = List(error.fieldName, error.errorType, error.typ)
+
+    def diffErrorLists(before: List[NodeValidationError], after: List[NodeValidationError]) = {
+      val errorsBefore = before.map(errorToKey).toSet
+      after.filterNot(error => errorsBefore.contains(errorToKey(error)))
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/TestModelMigrations.scala:58]: Can't we just use Set's difference instead?

File: components/openapi/src/test/scala/pl/touk/nussknacker/openapi/BaseOpenAPITest.scala (hunk 1 above)
Comment (piotrp) [reply components/openapi/src/test/scala/pl/touk/nussknacker/openapi/BaseOpenAPITest.scala:39]: Out of scope of this change, that was supposed to be a quick fix, without doing changes in project dependencies (`ResourceLoader` is not available in this module)

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/TestModelMigrations.scala [hunk 3]
```diff
@@ -50,30 +50,36 @@ class TestModelMigrations(migrations: ProcessingTypeDataProvider[ProcessMigratio
   }
 
   private def extractNewErrors(before: ValidationResult, after: ValidationResult) : ValidationResult = {
+    // simplified comparison key: we ignore error message and description
+    def errorToKey(error: NodeValidationError) = List(error.fieldName, error.errorType, error.typ)
```
[the remark this answers is not in the document]
Comment (piotrp) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/TestModelMigrations.scala]: Changed

File: utils/utils-internal/src/main/scala/pl/touk/nussknacker/engine/util/UriUtils.scala [hunk 4]
```diff
@@ -9,7 +10,7 @@ import java.net.{URLDecoder, URLEncoder}
   */
 object UriUtils {
 
-  private final val Charset = "UTF-8"
+  private final val Charset = StandardCharsets.UTF_8
```
[the remark this answers is not in the document]
Comment (piotrp) [reply utils/utils-internal/src/main/scala/pl/touk/nussknacker/engine/util/UriUtils.scala]: I don't know, you tell me - it's your code :P

(changed)

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/TestModelMigrations.scala (hunk 2 above)
Comment (arkadius) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/TestModelMigrations.scala:58]: Ehh right

**Pull request merged**